src/mirco_evaluate.cpp

```diff
@@ -12,16 +12,17 @@
 #include <string>
 #include <vector>
 using namespace std;
```

Comment:
I should not use this without limiting its scope. In this particular case, it does not matter that much, because it is in a cpp-file. However, if such a statement shows up in a header file, then it propagates through header including. 

I recommend to never use this outside tiny scopes to make sure that it never propagates. In fact, I'd never write `using namespace std`. In most cases, it's sufficient to list the two or three objects from the STL that you really want to use, e.g.
```c++
using std:cout
using std::vector
```
but only inside each function, where you want to use them (not a global scope).